Upstream-Status: Inappropriate [configuration]

On an Fedora host with glib-config installed pkgconfig-native can get into a
horrible state with recursive calls between pkg-config and glib-config.
The patch adds a configure time option to disable legacy script support in
pkgconfig with the --disable-legacy-scripts option, to maintain compatibility
the default is to leave the scripts enabled.

JL - 22/06/10
Index: pkg-config-0.23/configure.in
===================================================================
--- pkg-config-0.23.orig/configure.in	2008-01-16 22:48:07.000000000 +0000
+++ pkg-config-0.23/configure.in	2010-06-22 13:05:58.951984140 +0100
@@ -125,6 +125,14 @@
   AC_CONFIG_SUBDIRS(glib-1.2.10)
 fi # !native_win32
 
+# legacy *-configure scripts can cause headaches, add option to disable
+AC_ARG_ENABLE(legacy-scripts,
+        [AC_HELP_STRING([--enable-legacy-scripts=@<:@no/yes@:>@],
+                        [Whether pkg-config will try and use legacy scripts such as glib-config and gnome-config @<:@default=yes@:>@])],
+        [],
+        [enable_legacy=yes])
+AM_CONDITIONAL([LEGACY_SCRIPTS], [test x$enable_legacy = xyes])
+
 AC_FUNC_ALLOCA
 
 AC_CHECK_FUNCS(setresuid setreuid,break)
Index: pkg-config-0.23/parse.c
===================================================================
--- pkg-config-0.23.orig/parse.c	2008-01-16 20:42:49.000000000 +0000
+++ pkg-config-0.23/parse.c	2010-06-22 13:09:10.410129471 +0100
@@ -1195,6 +1195,11 @@
    * messages.
    */
   return NULL;
+#elif defined(LEGACY_SCRIPTS)
+  /* There are scenarios where we might not want to use these legacy
+   * scripts even if they are available.
+   */
+  return NULL;
 #else
 
   Package *pkg;
